### PR TITLE
test(program rules): RulesUtilsProvider unit tests

### PR DESCRIPTION
**Pull request by Balcan**
- Added unit tests to RulesUtilsProvider
- Added TextUtils to test\java\android\text

[ANDROAPP-2498](https://jira.dhis2.org/browse/ANDROAPP-2498)

### CODE REVIEW

File: app/src/main/java/org/dhis2/utils/RulesUtilsProviderImpl.kt [hunk 1]
```diff
@@ -145,19 +137,21 @@ class RulesUtilsProviderImpl(private val codeGenerator: CodeGenerator) : RulesUt
...
         fieldViewModels: MutableMap<String, FieldViewModel>,
-        rulesActionCallbacks: RulesActionCallbacks
+        rulesActionCallbacks: RulesActionCallbacks,
+        data: String
     ) {
         val model = fieldViewModels[showError.field()]
 
         if (model != null) {
-            fieldViewModels[showError.field()] = model.withError(showError.content())
+            fieldViewModels[showError.field()] = model.withError(showError.content() + " " + data)
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/utils/RulesUtilsProviderImpl.kt]: With kotlin we can use String interpolation to concatenate strings

`fieldViewModels[showError.field()] = model.withError("${showError.content()} $data")`

when you have to use a method or property of the variable it needs to be wrapped with the { } signs.

File: app/src/test/java/org/dhis2/utils/RulesUtilsProviderImplTest.kt [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+            actionCallbacks
+        )
+
+        verify(actionCallbacks, times(1)).setHideProgramStage(testingUid)
+    }
+
+    @Test
+    fun `RuleActionHideProgramStage should remove stage from possible selections`() {
+
+    }
```
Comment (JaimeToca) [inline app/src/test/java/org/dhis2/utils/RulesUtilsProviderImplTest.kt]: empty? :D

Comment (Balcan) [reply app/src/test/java/org/dhis2/utils/RulesUtilsProviderImplTest.kt]: :O

File: app/src/main/java/org/dhis2/utils/RulesUtilsProviderImpl.kt (hunk 1 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/utils/RulesUtilsProviderImpl.kt]: Done

**Pull request merged**